test/utils.test.js

```diff
@@ -453,4 +455,235 @@ describe('utils', function() {
+      // therefore we have to use `oneOf` here
+      var fn = actual.h;
+      delete actual.h;
+
+      assert.deepEqual(actual, expected);
+      assert.oneOf(fn, ['[Function: bar]', '[Function]']);
+    });
+
+    it('should reduce depth if payload size was exceeded', function() {
+      var actual = serializeException(
+        {
+          a: {
+            a: '50kB worth of payload pickle rick',
+            b: '50kB worth of payload pickle rick'
+          },
+          b: '50kB worth of payload pickle rick'
+        },
+        2,
+        100
+      );
+      var expected = {
+        a: '[Object]',
+        b: '50kB worth of payload pickle rick'
+      };
+
+      assert.deepEqual(actual, expected);
+    });
+
+    it('should reduce depth only one level at the time', function() {
+      var actual = serializeException(
+        {
+          a: {
+            a: {
+              a: {
+                a: [
+                  '50kB worth of payload pickle rick',
+                  '50kB worth of payload pickle rick',
+                  '50kB worth of payload pickle rick'
+                ]
+              }
+            },
+            b: '50kB worth of payload pickle rick'
+          },
+          b: '50kB worth of payload pickle rick'
+        },
+        4,
+        200
+      );
+      var expected = {
+        a: {
+          a: {
+            a: {
+              a: '[Array]'
+            }
+          },
+          b: '50kB worth of payload pickle rick'
+        },
+        b: '50kB worth of payload pickle rick'
+      };
+
+      assert.deepEqual(actual, expected);
+    });
+
+    it('should fallback to [Object] if cannot reduce payload size enough', function() {
+      var actual = serializeException(
+        {
+          a: '50kB worth of payload pickle rick',
+          b: '50kB worth of payload pickle rick',
+          c: '50kB worth of payload pickle rick',
+          d: '50kB worth of payload pickle rick'
+        },
+        1,
+        100
+      );
+      var expected = '[Object]';
+
+      assert.deepEqual(actual, expected);
+    });
+  });
+
+  describe('serializeKeysForMessage', function() {
+    it('should fit as many keys as possible in default limit of 40', function() {
+      var actual = serializeKeysForMessage([
+        'pickle',
+        'rick',
+        'morty',
+        'snuffles',
+        'server',
+        'request'
+      ]);
+      var expected = 'pickle, rick, morty, snuffles, server\u2026';
+      assert.equal(actual, expected);
+    });
+
+    it('shouldnt append ellipsis if have enough space', function() {
+      var actual = serializeKeysForMessage(['pickle', 'rick', 'morty']);
+      var expected = 'pickle, rick, morty';
+      assert.equal(actual, expected);
+    });
+
+    it('should default to no-keys message if empty array provided', function() {
+      var actual = serializeKeysForMessage([]);
+      var expected = '[object has no keys]';
```

Comment:
I thought about it, but I think it may be slightly confusing, as it looks like something coming from the browser itself and people may start to google for it :D

Reply:
`[Object (no keys)]`
`[Object with no keys]`

?

I don't know what kind of precedent we could follow here, but I think people might google for it either way (and ideally they'd find this repo?). I don't think this is /that/ important since it shouldn't be super common, I'm just throwing out other options

Reply:
oh, i didn't realize raven-node uses it already, that's good enough for me